### PR TITLE
⚡️ Performance improvement when rendering ImageNode

### PR DESCRIPTION
**Pull request by gsouquet**
Hi, 

First of all, thank you very much for `VideoContext`. It's great!

I have noticed some poor performances on "low-end" or older laptop when rendering an `ImageNode`.
Running a performance audit in the devtools would show something like: 

screen shot 2018-09-10 at 21 01 44

The `ImageNode` starts playing when the "green" bit gets bigger. In the Chrome devtools the "green" part of the chart refers to a painting operation.

If we have a closer look at the call stack, we can see an "Image Decode" event that takes place everytime that `gl.texImage2D` is called.

screen shot 2018-09-10 at 21 02 00

After digging a bit more on [gl.texImage2D](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/texImage2D) I noticed that we can draw on a texture using an [ImageBitmap](https://developer.mozilla.org/en-US/docs/Web/API/ImageBitmap) as a source.

> The ImageBitmap interface represents a bitmap image which can be drawn to a <canvas> without undue latency. It can be created from a variety of source objects using the createImageBitmap() factory method. ImageBitmap provides an asynchronous and resource efficient pathway to prepare textures for rendering in WebGL.

I have then changed the `ImageNode` class a bit, and after running some tests the timeline does less painting operations as the "Image Decode" step is completely gone from the call stack. 

screen shot 2018-09-10 at 20 58 57

It looks like this new Image source has some sporadic support. I have decided to gracefully degrade to use `HTMLImageElement` if the user agent does not support `ImageBitmap`.

Keen to hear your thoughts on that, I'd be happy to answer your questions if you have any 👍

### CODE REVIEW

**Comment by PTaylour** [comment]
Fantastic! This is great :)

**Comment by gsouquet** [comment]
@PTaylour yeah I have that checkbox ticked! Thank you for adding the tests

**Comment by PTaylour** [comment]
@reecelucas added you to review the tests I've just added here https://github.com/bbc/VideoContext/pull/103/commits/b4db86145af4c4cf2bfaf5b7b2e6210f3096ac0e

Couldn't get them to pass consistently locally. They were sometimes timing out on the `goto page` function. Have you seen this? Or likely an env thing?

File: src/SourceNodes/imagenode.js [hunk 1]
```diff
@@ -19,31 +19,41 @@ class ImageNode extends SourceNode {
...
         }
         if (this._isResponsibleForElementLifeCycle) {
             super._load();
-            this._element = new Image();
-            this._element.setAttribute("crossorigin", "anonymous");
-            this._element.src = this._elementURL;
-            this._element.onload = () => {
+            this._image = new Image();
+            this._image.setAttribute("crossorigin", "anonymous");
+            this._image.src = this._elementURL;
```
[the remark this answers is not in the document]
Comment (gsouquet) [reply src/SourceNodes/imagenode.js]: Awesome, just updated the PR!

**Comment by PTaylour** [comment]
regression tests now seem to be robust enough to merge this. Shall fire off a release now

**Pull request merged**